### PR TITLE
refactor top bar, call it consistently

### PR DESCRIPTION
**Pull request by anlhu**
New file for top bar, call it from each view

Event classes must implement some new methods to make the buttons work

gestures for events dont seem to work but that is not related to top bar

### CODE REVIEW

File: lib/common/view/topbar.dart [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+    backgroundColor: Colors.white,
+    shape: const RoundedRectangleBorder(
+        borderRadius: BorderRadius.only(
+      bottomLeft: Radius.circular(20),
+      bottomRight: Radius.circular(20),
+    )),
+    leading: IconButton(
+      icon: const Icon(Icons.calendar_month_rounded, color: Colors.black),
+      onPressed: () {
+        state.selectDate();
```
Comment (anlhu) [inline lib/common/view/topbar.dart:22]: Event will have to support this

File: lib/common/view/topbar.dart [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+                        builder: (context) => SingleDay(state.today),
+                      ),
+                    );
+                  } else {
+                    Navigator.of(context).push(
+                      MaterialPageRoute(
+                        builder: (context) => const TaskView(),
+                      ),
+                    );
+                  }
```
Comment (anlhu) [inline lib/common/view/topbar.dart:60]: not consistent, is there a way to remember the day?

File: lib/common/view/topbar.dart [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+      ],
+    ),
+    actions: <Widget>[
+      IconButton(
+        icon: const Icon(
+          Icons.search,
+          color: Colors.black,
+        ),
+        onPressed: () {
+          state.showSearchBar(context);
```
Comment (anlhu) [inline lib/common/view/topbar.dart:81]: Event will have to support this

File: lib/common/view/topbar.dart [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+            },
+          ),
+        ),
+        const SizedBox(width: 5),
+        Expanded(
+          child: IconButton(
+            icon: const Icon(Icons.calendar_month_rounded, color: Colors.black),
+            onPressed: () {
+              state.datePicker();
+            },
```
Comment (anlhu) [inline lib/common/view/topbar.dart:114]: event will have to support these

File: lib/common/view/topbar.dart [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+      IconButton(
+        icon: const Icon(Icons.search, color: Colors.black),
+        onPressed: () {
+          //showSearchBar();
+        },
+      ),
+      IconButton(
+        icon: const Icon(Icons.arrow_forward, color: Colors.black),
+        onPressed: () {
+          state.loadNextWeek();
```
Comment (anlhu) [inline lib/common/view/topbar.dart:181]: event will have to support this

**Review by lxu03 — APPROVED** [review]

**Pull request merged**